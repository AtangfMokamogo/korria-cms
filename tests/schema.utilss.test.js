const {
  validateReqSchema,
  validateParcelTypeSchema,
} = require('../utils/schema.utils');

describe('validateReqSchema function', () => {
  it('should return valid when all required keys are present', async () => {
    const req = { body: { key1: 'value1', key2: 'value2' } };
    const validKeys = ['key1', 'key2'];
    const result = await validateReqSchema(req, validKeys);
    expect(result).toMatchSnapshot();
  });

  it('should return missing keys when some required keys are not present', async () => {
    const req = { body: { key1: 'value1' } };
    const validKeys = ['key1', 'key2'];
    const result = await validateReqSchema(req, validKeys);
    expect(result).toMatchSnapshot();
  });

  it('should handle errors and return a code of -1', async () => {
    const req = { body: null };
    const validKeys = ['key1', 'key2'];
    const result = await validateReqSchema(req, validKeys);
    expect(result).toMatchSnapshot();
  });
});

describe('validateParcelTypeSchema function', () => {
  it('should return valid for a valid schema with image type', async () => {
    const req = {
      body: {
        schema: {
          fields: [
            { type: 'image', alt: 'Alt Text', src: 'image.jpg' },
          ],
        },
      },
    };
    const result = await validateParcelTypeSchema(req);
    expect(result).toMatchSnapshot();
  });

  it('should return valid for a valid schema with text type', async () => {
    const req = {
      body: {
        schema: {
          fields: [
            { fieldtype: 'text', content: { title: 'Title', payload: 'Text content' } },
          ],
        },
      },
    };
    const result = await validateParcelTypeSchema(req);
    expect(result).toMatchSnapshot();
  });

  it('should handle errors and return a code of -1', async () => {
    const req = { body: { schema: { fields: null } } };
    const result = await validateParcelTypeSchema(req);
    expect(result).toMatchSnapshot();
  });
});
