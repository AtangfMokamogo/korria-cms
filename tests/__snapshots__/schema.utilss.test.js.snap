// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validateParcelTypeSchema function should handle errors and return a code of -1 1`] = `
{
  "code": -1,
  "message": "Invalid, content type data missing in the schema",
}
`;

exports[`validateParcelTypeSchema function should return valid for a valid schema with image type 1`] = `
{
  "code": 1,
  "message": "Field has no name and/or data-type defined",
}
`;

exports[`validateParcelTypeSchema function should return valid for a valid schema with text type 1`] = `
{
  "code": 0,
  "message": "Valid",
}
`;

exports[`validateReqSchema function should handle errors and return a code of -1 1`] = `undefined`;

exports[`validateReqSchema function should return missing keys when some required keys are not present 1`] = `
{
  "code": -1,
  "message": [
    "key2",
  ],
}
`;

exports[`validateReqSchema function should return valid when all required keys are present 1`] = `
{
  "code": 0,
  "message": "valid",
}
`;
